xarray/core/groupby.py

```diff
@@ -102,8 +102,28 @@ def __init__(self, obj, group, squeeze=False, grouper=None):
         from .dataset import as_dataset
 
         if group.ndim != 1:
-            # TODO: remove this limitation?
-            raise ValueError('`group` must be 1 dimensional')
+            # try to stack the dims of the group into a single dim
+            # TODO: figure out how to exclude dimensions from the stacking
+            #       (e.g. group over space dims but leave time dim intact)
+            orig_dims = group.dims
+            stacked_dim_name = 'stacked_' + '_'.join(orig_dims)
+            # the copy is necessary here
+            group = group.stack(**{stacked_dim_name: orig_dims}).copy()
+            # without it, an error is raised deep in pandas
+            ########################
+            # xarray/core/groupby.py
+            # ---> 31     inverse, values = pd.factorize(ar, sort=True)
+            # pandas/core/algorithms.pyc in factorize(values, sort, order, na_sentinel, size_hint)
+            # --> 196     labels = table.get_labels(vals, uniques, 0, na_sentinel, True)
+            # pandas/hashtable.pyx in pandas.hashtable.Float64HashTable.get_labels (pandas/hashtable.c:10302)()
+            # pandas/hashtable.so in View.MemoryView.memoryview_cwrapper (pandas/hashtable.c:29882)()
+            # pandas/hashtable.so in View.MemoryView.memoryview.__cinit__ (pandas/hashtable.c:26251)()
+            # ValueError: buffer source array is read-only
```

Comment:
Probably better to move this traceback to a followup issue and just reference it by number in the code. Otherwise it's pretty distracting.